packages/core/elasticsearch/core-elasticsearch-client-server-internal/src/agent_manager.ts

```diff
@@ -53,13 +46,7 @@ export class AgentManager {
     return (connectionOpts: ConnectionOptions): NetworkAgent => {
       if (connectionOpts.url.protocol === HTTPS) {
         if (!httpsAgent) {
-          const config = Object.assign(
-            {},
-            DEFAULT_CONFIG,
-            this.agentOptions,
-            agentOptions,
-            connectionOpts.tls
-          );
```

Comment:
Not sure to get the exact implications of the changes in this file?

Reply:
we had a way to create a default agent config passed into the agent manager constructor. But this functionality was never used and sometimes confused my while developing cause it wasn't obvious where the config of the final agent actually came from. It was just a small moment of "what's going on here" but it felt like if we're not using this "feature" then it's easier to reason about the code if we just remove it altogether.

Reply:
actually there was several levels:
1. `DEFAULT_CONFIG` static defaults in code
2. `this.agentOptions` optional constructor paramater that can set defaults for any agent created by the agent factories of this agent manager
3. finally the kibana.yml defaults that are passed into the agent factories when they are created.

So now we just have (3)

Reply:
Makes sense! (1)'s values are equivalent to NodeJs's defaults, and (2) was not used.